engines/sci/engine/script_patches.cpp

```diff
@@ -3684,8 +3684,38 @@ static const uint16 phant1RatPatch[] = {
 	PATCH_END
 };
 
+// In Phantasmagoria the cursor’s hover state will not trigger on any of the buttons 
+// in the main menu after returning to the main menu from a game, or
+// when choosing "Quit" on the main menu and then cancelling the quit in the 
+// confirmation dialogue, until another button has been hovered and unhovered once.
+// This happens because the quit confirmation dialogue creates its own event handling loop 
+// which prevents the main event loop from handling the cursor leaving the button 
+// (which would reset global 193 to 0), and the dialogue does not reset global 193 itself, 
+// so it remains at 2 until a new button gets hovered and unhovered. 
+// There is not enough space in the confirmation dialogue code to add a reset of global 193, 
+// so we just remove the check entirely, since it is only used to avoid resetting the cursor’s 
+// view on every mouse movement, and this button type is only used 
+// on the main menu and the in-game control panel.
+// Applies to at least: English CD
+static const uint16 phant1RedQuitCursorSignature[] = {
+	0x18,                         // not
+	SIG_MAGICDWORD,
+	0x31, 0x25,                   // bnt $25
+	0x89, 0xc1,                   // lsg $c1
+	0x35, 0x02,                   // ldi 02
+	0x1c,                         // ne?
+	SIG_END
+};
+
+static const uint16 phant1RedQuitCursorPatch[] = {
+	PATCH_ADDTOOFFSET(+1),        // skip not
+	0x33, 0x07,                   // jmp [past global193 check]
```

Comment:
This patch is still overwriting the jump instruction for the _previous_ condition, so this earlier advice still applies:

> This patch is overwriting the end of the `((not (gDButton respondsTo: #exitDir))` check, because the signature starts at `not, bnt`, not at `lsg c1`. Either use the `PATCH_ADDTOOFFSET` macro to forward to the correct place to apply the patch (relative to the start of the signature), or remove the unnecessary extra bytes in the signature prior to the actual place we want to rewrite.

`bnt $25` is the end of the previous check. It is not the start of the global 193 check.

Nowhere else in this script is there an `lsg $c1, ldi 2`, and the patch requires only 2 bytes, so the signature does not need to be any longer than those two instructions. Therefore, the forwarding with `PATCH_ADDTOOFFSET` becomes unnecessary here since the patch will start exactly where we want to replace with a jump.